addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -199,20 +248,184 @@ def delete_bill(self, bill_id):
         except Exception as e:
             return "Failed to delete bill: {}".format(e)
 
+    @api.model
+    def create_bill_payment(self, bill_ids, journal_id, payment_date, payment_method_line_id):
+        AccountMove = self.env['account.move']
+        bills = AccountMove.browse(bill_ids)
+
+        if not bills.exists():
+            raise UserError("No valid bills found with the provided IDs.")
+        if any(bill.state != 'posted' for bill in bills):
+            raise UserError("All bills must be posted to proceed with payment.")
+
+        payment_register = self.env['account.payment.register'].with_context({
+            'active_model': 'account.move',
+            'active_ids': bill_ids,
+        }).create({
+            'payment_date': payment_date,
+            'journal_id': journal_id,
+            'payment_method_line_id': payment_method_line_id,
+        })
 
+        payments = payment_register.action_create_payments()
+
+        payment_data_list = []
+
+        for bill in bills:
+            # Directly accessing move lines and their full_reconcile_id
+            for move_line in bill.line_ids:
+                if move_line.full_reconcile_id:
+                    # Fetch other move lines that share the same full_reconcile_id and have a non-null payment_id
+                    reconciled_move_lines = self.env['account.move.line'].search([
+                        ('full_reconcile_id', '=', move_line.full_reconcile_id.id),
+                        ('payment_id', '!=', False),
+                        ('reconciled', '=', True),
+                    ])
+
+                    for r_move_line in reconciled_move_lines:
+                        payment_data = {
+                            'id': r_move_line.payment_id.id,  # ID of the payment
+                            'bill.id': bill.id,  # ID of the bill move associated with the payment
+                            'amount': r_move_line.debit if r_move_line.debit else 0.0,  # Credit amount
+                            'date': r_move_line.date or '',  # Date of the move line
```

Comment:
Is this the date of bill or date of bill payment? Can  you bring bill payment date too